realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -85,7 +85,7 @@ public void onSchemaVersionChanged(long currentVersion) {
                                 RealmCache.updateSchemaCache((Realm) BaseRealm.this);
                             }
                         }, true);
-        this.schema = new RealmSchema(this);
+        this.schema = new StandardRealmSchema(this);
```

Comment:
I think it should be.  I tried to do it and the number of changes required was just huge.  I'm leaving myself a note that this *does* need to happen.  If you can stand putting it off for a bit, we can get this hunk in...